### PR TITLE
MBS-11805 (I): Flow types and refactoring for guess case code (first half)

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11805

The main goal of this is to add flow types to guess case, and fix any issues that surfaces, which I expect will be a bunch. Also fixing smaller eslint issues and whatnot as I see them.

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, retest this please

**Comment by mwiencek** [comment]
@brainzbot reset this please

**Comment by mwiencek** [comment]
@brainzbot retest this please

**Comment by mwiencek** [comment]
@brainzbot test this please

File: root/static/scripts/guess-case/modes.js [hunk 1]
```diff
@@ -355,6 +355,7 @@ export const French: GuessCaseModeT = {
   name: 'French',
 
   runPostProcess(inputString) {
+    console.log('running French postprocess');
```
Comment (mwiencek) [inline root/static/scripts/guess-case/modes.js:358]: :scissors:

**Pull request merged**